OpenSSL and deprecated calls

What is the minimal requirement for the OpenSSL library version? I.e. is version 1.1.x still worth supporting? If not then `ERR_remove_state` can be removed. This would solve the following warning:

```
/mssl.c:39:9: warning: ‘ERR_remove_state’ is deprecated: Since OpenSSL 1.0.0 [-Wdeprecated-declarations]
   39 |         ERR_remove_state(0);
      |         ^~~~~~~~~~~~~~~~
In file included from /usr/include/openssl/engine.h:31,
                 from mssl.c:15:
/usr/include/openssl/err.h:476:30: note: declared here
  476 | OSSL_DEPRECATEDIN_1_0_0 void ERR_remove_state(unsigned long pid);
      |                              ^~~~~~~~~~~~~~~~

```

--- a/mssl.c
+++ b/mssl.c
@@ -36,8 +36,6 @@ void shutdown_ssl(void)
 
 	ERR_free_strings();
 
-	ERR_remove_state(0);
-	ENGINE_cleanup();
 	CONF_modules_free();
 	EVP_cleanup();
 	CRYPTO_cleanup_all_ex_data();

